codegen-renderers/src/main/kotlin/io/vrap/codegen/languages/java/model/second/JavaObjectTypeClassRenderer.kt

```diff
@@ -0,0 +1,150 @@
+                |import org.apache.commons.lang3.builder.EqualsBuilder;
+                |import org.apache.commons.lang3.builder.HashCodeBuilder;
+                |import org.apache.commons.lang3.builder.ToStringBuilder;
+                |import org.apache.commons.lang3.builder.ToStringStyle;
+                |import java.util.List;
+                |
+                |<${type.toComment().escapeAll()}>
+                |<${type.subTypesAnnotations()}>
+                |<${JavaSubTemplates.generatedAnnotation}>
+                |public class ${vrapType.simpleClassName}Impl ${type.type?.toVrapType()?.simpleName()?.let { "extends ${it}Impl" } ?: ""} implements ${vrapType.simpleClassName} {
+                |
+                |    <${type.toBeanFields().escapeAll()}>
+                |
+                |    <${type.setters().escapeAll()}>
+                |
+                |    <${type.getters().escapeAll()}>
+                |
+                |    @Override
+                |    public String toString() {
+                |        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
+                |    }
+                |
+                |    @Override
+                |    public boolean equals(Object o) {
+                |        return EqualsBuilder.reflectionEquals(this, o);
+                |    }
+                |
+                |    @Override
+                |    public int hashCode() {
+                |        return HashCodeBuilder.reflectionHashCode(this);
+                |    }
+                |}
+        """.trimMargin().keepIndentation()
+
+        return TemplateFile(
+            relativePath = "${vrapType.`package`}.${vrapType.simpleClassName}Impl".replace(".", "/") + ".java",
+            content = content
+        )
+    }
+
+    fun Property.toJavaField(): String {
+
+        return if (this.isPatternProperty()) {
+            "private Map<String,${this.type.toVrapType().simpleName()}> values;"
+        } else {
+            "private ${this.type.toVrapType().simpleName()} ${if (this.isPatternProperty()) "values" else this.name};"
+        }
+
+    }
+
+    fun ObjectType.toBeanFields() = this.properties
+        .filter { it.name != this.discriminator }
+        .map { it.toJavaField() }.joinToString(separator = "\n\n")
+
+    fun ObjectType.setters() = this.properties
+        //Filter the discriminators because they don't make much sense the generated bean
+        .filter { it.name != this.discriminator }
+        .map { it.setter() }
+        .joinToString(separator = "\n\n")
+
+
+    fun ObjectType.getters() = this.properties
+        //Filter the discriminators because they don't make much sense the generated bean
+        .filter { it.name != this.discriminator }
+        .map { it.getter() }
+        .joinToString(separator = "\n\n")
+
+
+    fun Property.setter(): String {
+        return if (this.isPatternProperty()) {
+
+            """
+            |@JsonAnySetter
```

Comment:
I think that we don't have to repeat these jackson annotations here and I would suggest to remove them.